--- origsrc/generatorrunner-0.6.7/CMakeLists.txt	2011-03-02 17:29:00.000000000 -0600
+++ src/generatorrunner-0.6.7/CMakeLists.txt	2011-03-24 23:37:07.784774900 -0500
@@ -14,7 +14,7 @@ if(MSVC)
     set(CMAKE_CXX_FLAGS "/Zc:wchar_t- /EHsc /DWIN32 /D_WINDOWS /D_SCL_SECURE_NO_WARNINGS")
 else()
     option(ENABLE_GCC_OPTIMIZATION "Enable specific GCC flags to optimize library size and performance. Only available on Release Mode" 0)
-    if (CMAKE_HOST_UNIX)
+    if (CMAKE_HOST_UNIX AND NOT CYGWIN)
         set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -fvisibility=hidden")
     endif()
     set(CMAKE_CXX_FLAGS_DEBUG "-g")
--- origsrc/generatorrunner-0.6.7/data/GeneratorRunnerConfig.cmake.in	2011-03-02 17:29:00.000000000 -0600
+++ src/generatorrunner-0.6.7/data/GeneratorRunnerConfig.cmake.in	2011-03-25 00:20:32.668190800 -0500
@@ -6,6 +6,8 @@
 SET(GENERATORRUNNER_INCLUDE_DIR "@CMAKE_INSTALL_PREFIX@/include/generatorrunner@generator_SUFFIX@")
 if(MSVC)
     SET(GENERATORRUNNER_LIBRARY "@LIB_INSTALL_DIR@/@CMAKE_SHARED_LIBRARY_PREFIX@genrunner@generator_SUFFIX@.lib")
+elseif(CYGWIN)
+    SET(GENERATORRUNNER_LIBRARY "@LIB_INSTALL_DIR@/@CMAKE_IMPORT_LIBRARY_PREFIX@genrunner@generator_SUFFIX@@CMAKE_IMPORT_LIBRARY_SUFFIX@")
 elseif(WIN32)
     SET(GENERATORRUNNER_LIBRARY "@CMAKE_INSTALL_PREFIX@/bin/@CMAKE_SHARED_LIBRARY_PREFIX@genrunner@generator_SUFFIX@@CMAKE_SHARED_LIBRARY_SUFFIX@")
 else()
--- origsrc/generatorrunner-0.6.7/generatorrunnermacros.h	2011-03-02 17:29:00.000000000 -0600
+++ src/generatorrunner-0.6.7/generatorrunnermacros.h	2011-03-25 00:14:26.447149600 -0500
@@ -25,7 +25,7 @@
 #define GENERATORRUNNERMACROS_H
 
 // GENRUNNER_API is used for the public API symbols.
-#if defined _WIN32 || defined __CYGWIN__
+#if defined _WIN32
     #define GENRUNNER_EXPORT __declspec(dllexport)
     #if GENRUNNER_EXPORTS
         #define GENRUNNER_API GENRUNNER_EXPORT
